File: leetcode/medium/4335-minimum-adjacent-swaps-to-partition-array/solution.java
class Solution {
    public int minAdjacentSwaps(int[] nums, int a, int b) {
        long ans = 0;
        long c2 = 0;
        long c1 = 0;
        for(int num  : nums){
            if(num > b){
                c2 ++;
            }
            else if (num >= a){
                ans += c2;
                c1++;
            }
            else{
                ans += c1 + c2;
            }
        }
        return (int) (ans % 1000000007);
    }
}

File: leetcode/medium/4335-minimum-adjacent-swaps-to-partition-array/solution.py
class Solution:
    def minAdjacentSwaps(self, nums: list[int], a: int, b: int) -> int:
        mod = 10**9 +7
        ans = 0
        c1 = 0
        c2 = 0

        for n in nums:
            if n > b:
                c2 += 1
            elif n >= a:
                ans += c2
                c1 += 1
            else:
                ans += c1 + c2

        return ans % mod